pkg/validation/admission.go

```diff
@@ -0,0 +1,72 @@
+package validation
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metal3v1alpha1 "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+)
+
+// ValidateHostForAdmission returns an error if the host fails uniqueness
+// validation, and cannot be added to the cluster.  This is ordinarily run by an
+// admission webhook.
+func ValidateHostForAdmission(c client.Client, host metal3v1alpha1.BareMetalHost) error {
```

Comment:
Once we have a validating webhook we'll almost certainly want to use it for other things that don't involve uniqueness across the whole cluster. It might pay to group this stuff that does in a function with a more specific name.